### PR TITLE
opensearch-dashboards 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
OpenSearch Dashboards — is a OpenSearch fork of Kibana. Formula based on `kibana` formula with some minor changes.

### CODE REVIEW

File: Formula/opensearch-dashboards.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  end
+
+  def caveats
+    <<~EOS
+      Config: #{etc}/opensearch-dashboards/
+      If you wish to preserve your plugins upon upgrade, make a copy of
+      #{opt_prefix}/plugins before upgrading, and copy it into the
+      new keg location after upgrading.
+    EOS
+  end
```
Comment (carlocab) [inline Formula/opensearch-dashboards.rb:48]: I know you inherited this from `kibana` but I wonder if we can convince this to install the plugins into `var`, as with `opensearch`:
https://github.com/Homebrew/homebrew-core/blob/f52a2bf4dccbfbc22ec0ec2d0998a083c90e1ab1/Formula/opensearch.rb#L58-L76

Comment (bayandin) [reply Formula/opensearch-dashboards.rb:48]: It looks like it's impossible to do 😞 
In all the places it's hardcoded to something like "<project root>/plugins". and it's not customisable. 
For example https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.0.0/src/cli_plugin/install/settings.js#L67

Comment (carlocab) [reply Formula/opensearch-dashboards.rb:48]: My understanding is that it's the same for `opensearch`, we just trick it with a symlink in order to make it put plug-ins in a static location.

Comment (bayandin) [reply Formula/opensearch-dashboards.rb]: You're right!
I've updated the formula, now it's inlined with `opensearch` a bit more.

File: Formula/opensearch-dashboards.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    (var/"opensearch-dashboards/plugins").mkpath
+    ln_s var/"opensearch-dashboards/plugins", prefix/"plugins" unless (prefix/"plugins").exist?
+
+    ln_s etc/"opensearch-dashboards", prefix/"config" unless (prefix/"config").exist?
+  end
+
+  def caveats
+    <<~EOS
+      Data:    #{var}/lib/opensearch-dashboards/
+      Logs:    #{var}/log/opensearch-dashboards/opensearch-dashboards.log
```
Comment (carlocab) [inline Formula/opensearch-dashboards.rb:50]: Do we also want to rename this as we do with the `opensearch` formula? Or do the reasons we did this there not apply here?

Comment (bayandin) [reply Formula/opensearch-dashboards.rb:50]: I don't think `opensearch-dashboards` has clusters, at least it's not in their config: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.0.0/config/opensearch_dashboards.yml

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**